src/Compilers/CSharp/Portable/Symbols/TypeSymbol.cs

```diff
@@ -1015,6 +1015,51 @@ private static Symbol ComputeImplementationForInterfaceMember(Symbol interfaceMe
             return implicitImpl;
         }
 
+        internal static MethodSymbol TryFindBaseImplementationInInterface(NamedTypeSymbol baseInterface, MethodSymbol method, DiagnosticBag diagnosticsOpt, SyntaxNode nodeOpt, Binder accessCheckBinderOpt)
+        {
+            Debug.Assert(baseInterface.IsInterface);
+            Debug.Assert(diagnosticsOpt is null == nodeOpt is null);
+            Debug.Assert(diagnosticsOpt is null == accessCheckBinderOpt is null);
+
+            HashSet<DiagnosticInfo> useSiteDiagnostics = null;
+            Symbol conflictingImplementation1 = null;
+            Symbol conflictingImplementation2 = null;
+            Symbol implementingMember = TypeSymbol.FindImplementationInInterface(method, baseInterface) ??
+                                        TypeSymbol.FindMostSpecificImplementation(method, baseInterface,
```

Comment:
It isn't clear how this handles reabstraction [https://github.com/dotnet/csharplang/blob/master/proposals/default-interface-methods.md#reabstraction](https://github.com/dotnet/csharplang/blob/master/proposals/default-interface-methods.md#reabstraction) .  I would expect this to be able to return an abstract override if that is most specific.  But the caller must be prepared to report that the most specific method is abstract.  Similarly, when one branch of a diamond is reabstracted and the other is concrete, there is no most specific override.  Do we have tests showing these scenarios in action?

Reply:
> It isn't clear how this handles reabstraction 

We do not support reabstraction.

---
In reply to: [255765763](https://github.com/dotnet/roslyn/pull/33281#discussion_r255765763) [](ancestors = 255765763)

Reply:
I see that reabstraction is on the TODO list [https://github.com/dotnet/roslyn/issues/17952](https://github.com/dotnet/roslyn/issues/17952)

---
In reply to: [255766000](https://github.com/dotnet/roslyn/pull/33281#discussion_r255766000) [](ancestors = 255766000,255765763)

Reply:
> I see that reabstraction is on the TODO list [https://github.com/dotnet/roslyn/issues/17952](https://github.com/dotnet/roslyn/issues/17952)

We are not planning to support it because the syntax (an explicit implementation) doesn't allow that. I will make a note in the issue.

---
In reply to: [255767363](https://github.com/dotnet/roslyn/pull/33281#discussion_r255767363) [](ancestors = 255767363,255766000,255765763)

Reply:
The intent is to extend the syntax to allow `abstract void I.M();`

---
In reply to: [255768706](https://github.com/dotnet/roslyn/pull/33281#discussion_r255768706) [](ancestors = 255768706,255767363,255766000,255765763)

Reply:
> The intent is to extend the syntax to allow abstract void I.M();

I am not aware of plans to do this for C# 8.

---
In reply to: [255775466](https://github.com/dotnet/roslyn/pull/33281#discussion_r255775466) [](ancestors = 255775466,255768706,255767363,255766000,255765763)

Reply:
This is required for compatibility with Java, which is one of our principal use cases.  I've added an open LDM issue for what syntax change we intend to support this.

---
In reply to: [255775466](https://github.com/dotnet/roslyn/pull/33281#discussion_r255775466) [](ancestors = 255775466,255768706,255767363,255766000,255765763)